lib/pass/dead-code-elimination.js

```diff
@@ -480,11 +480,13 @@
                 case Syntax.TryStatement:
                     live |= visit(node.block);
 
-                    if (node.handlers && node.handlers.length) {
+                    // NOTE: TryStatement interface changed from {handlers: [CatchClause]} to {handler: CatchClause}; we support both
+                    handlers = node.handlers || (node.handler ? [node.handler] : []);
+                    if (handlers && handlers.length) {
                         if (!status.revive()) {
                             status.insert(node);
                         }
-                        node.handlers = eliminate(node, node.handlers);
+                        handlers = eliminate(node, handlers);
```

Comment:
esmangle operates in one of two modes: mutate the input AST or deep copy the AST and mutate that. So no need to worry about it.